### PR TITLE
c42-fix-enum-types

### PR DESCRIPTION
**Pull request by TheOneHoff**
## Change Summary

- Removed the enum type from the database. The application now provides validation for fields

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The code works by "Smoke testing" (in other words, you have tested it your own and it works)
- [ ] The code has documentation (either comments or in main documentation platform)

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

### CODE REVIEW

File: app/app/models.py [hunk 1]
```diff
@@ -92,6 +88,27 @@ def validate_studentNumber(self, key, studentNumber):
     @validates('unitCode')
     def validate_unitCode(self, key, unitCode):
         for i, c in enumerate(unitCode):
-            if (i < 4 and not c.isupper()) or (i >= 4 and not c.isnumeric()):
+            if (i < 4 and not (c.isupper() or c.islower())) or (i >= 4 and not c.isnumeric()):
```
Comment (Alexkszzz) [inline app/app/models.py]: Works well!

File: app/app/models.py [hunk 2]
```diff
@@ -92,6 +88,27 @@ def validate_studentNumber(self, key, studentNumber):
...
+            raise ValueError('Enquiry is an invalid type')
+        else:
+            return queue
+
+    @validates('status')
+    def validate_status(self, key, status):
+        if status not in statusType:
+            raise ValueError('Enquiry is an invalid type')
+        else:
+            return status
```
Comment (Alexkszzz) [inline app/app/models.py]: Looks good! May consider to add validate_studentName as well to make sure there is no numeric character in it.

Comment (TheOneHoff) [reply app/app/models.py]: Good idea. Done!

File: app/migrations/versions/f63f9cf579c1_.py [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
+Create Date: 2021-09-20 08:27:20.487921
 
 """
 from alembic import op
 import sqlalchemy as sa
 
 
 # revision identifiers, used by Alembic.
-revision = '097d6eedce34'
+revision = 'f63f9cf579c1'
```
Comment (Alexkszzz) [inline app/migrations/versions/f63f9cf579c1_.py:13]: Error in migration. Wrong spelling

Comment (TheOneHoff) [reply app/migrations/versions/f63f9cf579c1_.py:13]: Good catch. Latest migration should have fixed the issue

**Review by JordanHartley — APPROVED** [review]
All looks fine!

File: app/app/models.py [hunk 4]
```diff
@@ -80,6 +82,14 @@ def __init__(self, studentName, studentNumber, unitCode, enquiry, queue, status,
...
         return f"<Queue #{self.id}{{Name: {self.studentName}, ID: {self.studentNumber}, Unit: {self.unitCode}, Enquiry: {self.enquiry}, Queue: {self.queue}>"
 
+    @validates('studentName')
+    def validates_studentName(self, key, studentName):
+        for c in studentName:
+            if c.isnumeric() or c in invalidChar:
+                raise ValueError("Invalid character in studentName")
+
+        return studentName
+
```
Comment (Alexkszzz) [inline app/app/models.py:92]: Looks good to me!

File: app/migrations/versions/8408aefd8f90_.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
+"""empty message
+
+Revision ID: 8408aefd8f90
+Revises: 097d6eedce34
+Create Date: 2021-09-21 04:58:27.495271
+
+"""
```
Comment (Alexkszzz) [inline app/migrations/versions/8408aefd8f90_.py:7]: new migration has fixed the 'undefined' when choosing 'Grammar' or 'Research Proposal'. Tested and it worked!

**Review by Alexkszzz — APPROVED** [review]

**Comment by frinzekt** [comment]
If this has been reviewed, and ready to go, don't let the PR go stale here. Merge it

The longer a branch is away from main branch. The higher the likelyhood of it causing merge conflicts because most branches are branching of master (or develop) which means always keep it updated. https://softwareengineering.stackexchange.com/questions/395021/is-it-better-to-merge-often-or-only-after-completion-do-a-big-merge-of-feature

**Pull request merged**